lib/src/room.dart

```diff
@@ -1600,9 +1605,18 @@ class Room {
             .toList() ??
         [];
 
-    if (cache) {
+    if (cacheInMemory) {
       for (final user in users) {
-        setState(user); // at *least* cache this in-memory
+        setState(user);
+      }
+    }
+    if (cacheInDatabase) {
+      for (final user in users) {
+        await client.database?.storeEventUpdate(
```

Comment:
We already do as this is often called in a sync so it is in a batch there. If we put it into another batch we increase the problem with transaction-inside-transaction. Until we have found a better solution for this, we should try to avoid it